File: app/src/services/reviewService.js
import api from '../config/api';

export const reviewService = {
  createReview: async reviewData => {
    const response = await api.post('/reviews', reviewData);
    return response.data;
  },

  getSalonReviews: async (salonId, page = 1) => {
    const response = await api.get(`/reviews/salon/${salonId}`, {
      params: {page},
    });
    return response.data;
  },

  getMyReviews: async (page = 1) => {
    const response = await api.get('/reviews/my-reviews', {params: {page}});
    return response.data;
  },

  respondToReview: async (reviewId, responseText) => {
    const response = await api.post(`/reviews/${reviewId}/respond`, {
      responseText,
    });
    return response.data;
  },
};

File: luxtril_flutter/lib/services/review_service.dart
import 'api_client.dart';
import '../models/review.dart';

class ReviewService {
  final ApiClient _api = ApiClient();

  Future<Review> createReview(Map<String, dynamic> data) async {
    final response = await _api.post('/reviews', body: data);
    return Review.fromJson(response['review'] ?? response);
  }

  Future<List<Review>> getSalonReviews(String salonId) async {
    final response = await _api.get('/reviews/salon/$salonId');
    final List<dynamic> data = response['reviews'] ?? response['data'] ?? [];
    return data.map((r) => Review.fromJson(r)).toList();
  }

  Future<Review> respondToReview(String id, String responseText) async {
    final result = await _api.put('/reviews/$id/respond', body: {
      'response_text': responseText,
    });
    return Review.fromJson(result['review'] ?? result);
  }
}
